### PR TITLE
Add Webpack Snap example

### PR DESCRIPTION
**Pull request by Mrtenz**
This adds an example snap made with Webpack, and fixes some issues with the current Webpack plugin regarding types.

### CODE REVIEW

File: packages/plugin-webpack/src/plugin.ts [hunk 1]
```diff
@@ -1,12 +1,12 @@
 import { postProcessBundle, PostProcessOptions } from '@metamask/snap-utils';
-import { Compiler, WebpackPluginInstance } from 'webpack';
+import { Compiler } from 'webpack';
 import { RawSource } from 'webpack-sources';
 
 const PLUGIN_NAME = 'SnapsWebpackPlugin';
 
 export type Options = PostProcessOptions;
 
-export default class SnapsWebpackPlugin implements WebpackPluginInstance {
```
Comment (FrederikBolding) [inline packages/plugin-webpack/src/plugin.ts:9]: How come we don't need this anymore? 🤔

File: packages/plugin-webpack/package.json [hunk 2]
```diff
@@ -35,14 +35,13 @@
   "devDependencies": {
     "@metamask/auto-changelog": "^2.5.0",
     "@types/jest": "^27.4.1",
-    "@types/webpack": "^5.28.0",
```
Comment (FrederikBolding) [inline packages/plugin-webpack/package.json:38]: Are types included now?

File: packages/examples/examples/webpack/package.json [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  "main": "src/index.js",
+  "files": [
+    "dist/",
+    "snap.manifest.json"
+  ],
+  "scripts": {
+    "build": "webpack",
+    "manifest": "mm-snap manifest --fix",
+    "eval": "mm-snap eval -b dist/snap.js",
+    "serve": "concurrently \"webpack --watch\" \"http-server -c-1 .\""
```
Comment (FrederikBolding) [inline packages/examples/examples/webpack/package.json:20]: Is there a reason to not use `mm-snap serve`?

File: packages/examples/examples/webpack/webpack.config.ts [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+  },
+};
+
+// Configuration for the Snap bundle
+const snapConfig: Configuration = merge(common, {
+  entry: {
+    snap: './src/snap.ts',
+  },
+  plugins: [
+    new SnapsWebpackPlugin(),
```
Comment (FrederikBolding) [inline packages/examples/examples/webpack/webpack.config.ts:50]: Is this needed when it is already in `common`?

File: yarn.lock [hunk 5]
```diff
@@ -210,6 +210,29 @@ __metadata:
   languageName: node
   linkType: hard
 
+"@babel/core@npm:^7.17.10":
```
Comment (FrederikBolding) [inline yarn.lock:213]: Where does all the new dependencies come from?

File: packages/examples/examples/webpack/src/index.html [hunk 6]
```diff
@@ -0,0 +1,28 @@
+<!doctype html>
```
Comment (FrederikBolding) [inline packages/examples/examples/webpack/src/index.html:1]: Do you not need a template somewhere to inject the JS?

File: packages/examples/examples/webpack/package.json [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+  },
+  "devDependencies": {
+    "@babel/core": "^7.17.10",
+    "@babel/preset-typescript": "^7.16.7",
+    "@metamask/snap-types": "^0.14.0",
+    "@metamask/snaps-cli": "^0.14.0",
+    "@metamask/snaps-webpack-plugin": "^0.14.0",
+    "@types/node": "^17.0.33",
+    "@types/webpack-dev-server": "^4.7.2",
+    "assemblyscript": "^0.19.23",
```
Comment (FrederikBolding) [inline packages/examples/examples/webpack/package.json:30]: Why does this package have all these dependencies which I assume are unused?

File: packages/plugin-webpack/src/plugin.ts (hunk 1 above)
Comment (Mrtenz) [reply packages/plugin-webpack/src/plugin.ts:9]: It breaks the type for whatever reason, and it seems like other plugins (including the ones by Webpack) don't do it either.

File: packages/plugin-webpack/package.json (hunk 2 above)
Comment (Mrtenz) [reply packages/plugin-webpack/package.json:38]: Yes.

File: yarn.lock (hunk 5 above)
Comment (Mrtenz) [reply yarn.lock:213]: It uses `babel-loader` to process TS files

File: packages/examples/examples/webpack/webpack.config.ts (hunk 4 above)
Comment (Mrtenz) [reply packages/examples/examples/webpack/webpack.config.ts:50]: It's not in `common`, and shouldn't be used for the web config.

File: packages/examples/examples/webpack/package.json (hunk 7 above)
Comment (Mrtenz) [reply packages/examples/examples/webpack/package.json:30]: Copied the package.json from another example and forgot to remove those.

**Review by FrederikBolding — APPROVED** [review]

**Pull request merged**